axlearn/common/host_array_test.py

```diff
@@ -28,26 +28,24 @@ def is_supported(
     global_batch_size: int,
     data_partition: DataPartitionType,
 ):
-    return (
-        is_supported_platform(platform)
-        and is_supported_mesh_shape(mesh_shape)
-        and (
-            data_partition == DataPartitionType.REPLICATED
-            or global_batch_size % jax.device_count() == 0
-        )
-    )
+    if not is_supported_platform(platform):
+        return False, f'Platform "{platform}" not supported with devices {jax.devices()}.'
+    if not is_supported_mesh_shape(mesh_shape):
+        return False, f'Mesh shape "{mesh_shape}" not supported with device_count "{jax.device_count()}".'
+    if data_partition != DataPartitionType.REPLICATED:
+        return False, f'Data partition is "{data_partition}", expected "DataPartitionType.REPLICATED".'
+    if global_batch_size % jax.device_count() != 0:
+        return False, 'Global batch has to be divisible with number of devices. Global batch is "{global_batch_size}", number of devices is "{jax.device_count()}".'
+    return True , ""
 
 
 class HostArrayTest(TestCase):
     @parameterized.parameters(
-        filter(
-            lambda params: is_supported(*params),
-            itertools.product(
-                ("cpu", "tpu"),  # platform,
-                ((1, 1), (4, 1), (2, 2), (8, 1), (4, 2)),  # mesh_shape
-                (1, 16),  # global_batch_size
-                (DataPartitionType.FULL, DataPartitionType.REPLICATED),  # data_partition
-            ),
+        itertools.product(
```

Comment:
instead of filtering in @parameterized.parameters (which means tests are not skipped but disappear and can result in pytest error) I'm doing pytest.skip in the function (line 65)